### PR TITLE
Added request-driven cached test repo

### PR DESCRIPTION
**Pull request by mateuszmidor**
KNTK-221.
Test results cache is updated on request but only if older than configured seconds count.
Cache is per application instance; multiple instances can be run but each maintains individual cache.

### CODE REVIEW

File: domain/cached_repo_request_driven.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from datetime import datetime, timedelta
+
+from domain.model.mesh_results import MeshResults
+from domain.repo import Repo
+from domain.types import TestID
+
+logger = logging.getLogger(__name__)
+
+
+class CachedRepoRequestDriven:
```
Comment (Fenthick) [inline domain/cached_repo_request_driven.py:13]: Github doesn't show the diff :(

File: main.py [hunk 2]
```diff
@@ -7,48 +7,38 @@
...
+            self._cached_repo = CachedRepoRequestDriven(
+                repo,
+                config.test_id,
+                config.data_update_period_seconds,
+                config.data_update_lookback_seconds,
+            )
+            self._config = config
         except Exception as err:
-            logger.exception(f"WebApp initialization failure: {err}")
+            logger.exception("WebApp initialization failure")
```
Comment (Fenthick) [inline main.py:34]: Won't this change filter the root cause of the error?

Comment (mateuszmidor) [reply main.py:34]: Logger.exception attaches current exception to the log

File: domain/cached_repo_request_driven.py (hunk 1 above)
Comment (mateuszmidor) [reply domain/cached_repo_request_driven.py:13]: > Github doesn't show the diff :(
This is a new file, entire file is shown

Comment (Fenthick) [reply domain/cached_repo_request_driven.py:13]: I know, but the implementation is based on cached_repo.py

**Review by Fenthick — APPROVED** [review]

File: main.py [hunk 3]
```diff
@@ -7,48 +7,38 @@
...
 
-from domain.cached_repo import CachedRepo
+from domain.cached_repo_request_driven import CachedRepoRequestDriven
 from infrastructure.config import ConfigYAML
 from infrastructure.data_access.http.synthetics_repo import SyntheticsRepo
 from presentation.main_view import make_page_layout
 from presentation.matrix_view import make_mesh_test_matrix_layout
 
+FORMAT = "%(asctime)-15s  %(message)s"
+logging.basicConfig(level=logging.DEBUG, format=FORMAT)
```
Comment (mmac-m3a) [inline main.py:17]: Default log level should not be DEBUG. Make debugging (or log level in general) controllable via configuration.

**Review by mmac-m3a — APPROVED** [review]
LGTM

**Pull request merged**